### ISSUE TITLE
Running `git lab add-account` on a non-https server fails

### ISSUE THREAD

**Issue by mattiasb** [seveas/git-spindle#123]
Running `git lab add-account` for a host that's non-https while not defining the protocol gives an error like this:

```
 ~ $ git lab add-account --host=git.mycompany.se mc
GitLab user: mattiasb
GitLab password: 
Traceback (most recent call last):
  File "/home/mattiasb/.local/lib/python3.5/site-packages/requests/packages/urllib3/connection.py", line 142, in _new_conn
    (self.host, self.port), self.timeout, **extra_kw)
  File "/home/mattiasb/.local/lib/python3.5/site-packages/requests/packages/urllib3/util/connection.py", line 98, in create_connection
    raise err
  File "/home/mattiasb/.local/lib/python3.5/site-packages/requests/packages/urllib3/util/connection.py", line 88, in create_connection
    sock.connect(sa)
OSError: [Errno 113] No route to host

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/mattiasb/.local/lib/python3.5/site-packages/requests/packages/urllib3/connectionpool.py", line 595, in urlopen
    chunked=chunked)
  File "/home/mattiasb/.local/lib/python3.5/site-packages/requests/packages/urllib3/connectionpool.py", line 352, in _make_request
    self._validate_conn(conn)
  File "/home/mattiasb/.local/lib/python3.5/site-packages/requests/packages/urllib3/connectionpool.py", line 831, in _validate_conn
    conn.connect()
  File "/home/mattiasb/.local/lib/python3.5/site-packages/requests/packages/urllib3/connection.py", line 254, in connect
    conn = self._new_conn()
  File "/home/mattiasb/.local/lib/python3.5/site-packages/requests/packages/urllib3/connection.py", line 151, in _new_conn
    self, "Failed to establish a new connection: %s" % e)
requests.packages.urllib3.exceptions.NewConnectionError: <requests.packages.urllib3.connection.VerifiedHTTPSConnection object at 0x7fc1da982438>: Failed to establish a new connection: [Errno 113] No route to host

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/mattiasb/.local/lib/python3.5/site-packages/requests/adapters.py", line 423, in send
    timeout=timeout
  File "/home/mattiasb/.local/lib/python3.5/site-packages/requests/packages/urllib3/connectionpool.py", line 640, in urlopen
    _stacktrace=sys.exc_info()[2])
  File "/home/mattiasb/.local/lib/python3.5/site-packages/requests/packages/urllib3/util/retry.py", line 287, in increment
    raise MaxRetryError(_pool, url, error or ResponseError(cause))
requests.packages.urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='git.mycompany.se', port=443): Max retries exceeded with url: /api/v3/session (Caused by NewConnectionError('<requests.packages.urllib3.connection.VerifiedHTTPSConnection object at 0x7fc1da982438>: Failed to establish a new connection: [Errno 113] No route to host',))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/mattiasb/.local/lib/python3.5/site-packages/gitspindle/glapi.py", line 235, in rawPost
    timeout=self.timeout)
  File "/home/mattiasb/.local/lib/python3.5/site-packages/requests/api.py", line 110, in post
    return request('post', url, data=data, json=json, **kwargs)
  File "/home/mattiasb/.local/lib/python3.5/site-packages/requests/api.py", line 56, in request
    return session.request(method=method, url=url, **kwargs)
  File "/home/mattiasb/.local/lib/python3.5/site-packages/requests/sessions.py", line 475, in request
    resp = self.send(prep, **send_kwargs)
  File "/home/mattiasb/.local/lib/python3.5/site-packages/requests/sessions.py", line 596, in send
    r = adapter.send(request, **kwargs)
  File "/home/mattiasb/.local/lib/python3.5/site-packages/requests/adapters.py", line 487, in send
    raise ConnectionError(e, request=request)
requests.exceptions.ConnectionError: HTTPSConnectionPool(host='git.mycompany.se', port=443): Max retries exceeded with url: /api/v3/session (Caused by NewConnectionError('<requests.packages.urllib3.connection.VerifiedHTTPSConnection object at 0x7fc1da982438>: Failed to establish a new connection: [Errno 113] No route to host',))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/mattiasb/.local/bin/git-lab", line 11, in <module>
    GitLab().main()
  File "/home/mattiasb/.local/lib/python3.5/site-packages/gitspindle/__init__.py", line 344, in main
    func(opts)
  File "/home/mattiasb/.local/lib/python3.5/site-packages/gitspindle/__init__.py", line 357, in add_account
    self.login()
  File "/home/mattiasb/.local/lib/python3.5/site-packages/gitspindle/gitlab.py", line 47, in login
    self.gl.auth()
  File "/home/mattiasb/.local/lib/python3.5/site-packages/gitspindle/glapi.py", line 170, in auth
    self.credentials_auth()
  File "/home/mattiasb/.local/lib/python3.5/site-packages/gitspindle/glapi.py", line 177, in credentials_auth
    {'email': self.email, 'password': self.password})
  File "/home/mattiasb/.local/lib/python3.5/site-packages/gitspindle/glapi.py", line 238, in rawPost
    "Can't connect to GitLab server (%s)" % self._url)
gitspindle.glapi.GitlabConnectionError: Can't connect to GitLab server (https://git.mycompany.se/api/v3)
```

**Comment by mattiasb** [comment]
Running `git lab add-account --host=https://git.mycompany.se mc` works fine. 

I would expect either a message saying that I need to define protocol or that `git-spindle` tries both http and https (this might be insecure, so perhaps the former is preferred).

**Comment by seveas** [comment]
You're having fun filing issues :)

They all make sense, even the one you already closed. Will try to tackle them on sunday.

**Comment by mattiasb** [comment]
Great! :)

Yeah I was testing git-spindle against our gitlab instance at work and felt like I should just report whatever issues I found. :)

**Comment by seveas** [comment]
Please keep'em coming. As I still don't use gitlab at work, that part of the codebase isn't the best tested.

**Comment by mattiasb** [comment]
Cool! Hopefully I'll have time to fix some issues / add new features myself as I start using git-spindle more.

**Comment by seveas** [comment]
Fixed in 42d4c28
